test/e2e/common_suite_test.go

```diff
@@ -114,15 +120,58 @@ func (tc *testCase) run() {
 				if err = client.Resources().Create(ctx, tc.pod); err != nil {
 					t.Fatal(err)
 				}
-				if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-					t.Fatal(err)
+				if tc.podState != "" {
```

Comment:
`wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT))` should be equivalent to `wait.For(conditions.New(client.Resources()).PodPhaseMatch(tc.pod, v1.PodRunning)`, so I think we can remove the need to have this `if` `else` branch by

- Setting a default `podState` of `v1.PodRunning` in `newTestCase`
- Change the `wait.For` code to `wait.For(conditions.New(client.Resources()).PodPhaseMatch(tc.pod, tc.podState), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT))`, which covers test cases where the test expects a pod to reach the running phase and where the test expects a pod to reach the Succeeded phase.